sdk/client/src/stream/streaming_client.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{stream::traits::StreamingClientTransport, Error, Result};
+use diem_json_rpc_types::{stream::response::StreamJsonRpcResponse, Id};
+use std::{collections::HashMap, sync::Arc};
+use tokio::{
+    sync::{mpsc, RwLock},
+    task::JoinHandle,
+};
+
+use diem_types::event::EventKey;
+
+pub type StreamingClientReceiver = mpsc::Receiver<Result<StreamJsonRpcResponse>>;
+pub type StreamingClientSender = mpsc::Sender<Result<StreamJsonRpcResponse>>;
+pub type SubscriptionStreamResult = Result<(Id, StreamingClientReceiver)>;
```

Comment:
Its also not very ergonomic to have a subscription return a tuple, it would probably be better to return a type with wraps the stream and contains the Id of the stream, implementing Stream on the wrapper to make using it more ergonomic.